cvmfs/sync_mediator.cc

```diff
@@ -94,6 +95,28 @@ void SyncMediator::Add(SharedPtr<SyncItem> entry) {
     return;
   }
 
+  // .cvmfsbundles file type
+  if (entry->IsBundleSpec()) {
+    std::string json_string;
+
+    int fd = open(entry->GetUnionPath().c_str(), O_RDONLY);
+    if (fd < 0) {
+      PANIC(kLogStderr, "Could not open file: %s",
+          entry->GetUnionPath().c_str());
+    } else {
+      if (SafeReadToString(fd, &json_string)) {
```

Comment:
Let's reverse: `if (!SafeReadToString(...` then panic, in which case we again can drop the else branch.